File: snippets/fsharp/VS_Snippets_CLR_System/system.Byte.Parse/FS/parse.fs
open System
open System.Globalization

let callParse1 () =
    // <Snippet1>
    let stringToConvert = " 162"
    try
        let byteValue = Byte.Parse stringToConvert
        printfn $"Converted '{stringToConvert}' to {byteValue}."
    with
    | :? FormatException ->
        printfn $"Unable to parse '{stringToConvert}'."
    | :? OverflowException ->
        printfn $"'{stringToConvert}' is greater than {Byte.MaxValue} or less than {Byte.MinValue}."

    // The example displays the following output to the console:
    //       Converted ' 162' to 162.
    // </Snippet1>

let callParse2 () =
    // <Snippet2>
    let stringToConvert = " 214 "
    try
        let byteValue = Byte.Parse(stringToConvert, CultureInfo.InvariantCulture)
        printfn $"Converted '{stringToConvert}' to {byteValue}." 
    with
    | :? FormatException ->
        printfn $"Unable to parse '{stringToConvert}'."
    | :? OverflowException ->
        printfn $"'{stringToConvert}' is greater than {Byte.MaxValue} or less than {Byte.MinValue}." 

    let stringToConvert = " + 214 "
    try
        let byteValue = Byte.Parse(stringToConvert, CultureInfo.InvariantCulture)
        printfn $"Converted '{stringToConvert}' to {byteValue}." 
    with
    | :? FormatException ->
        printfn $"Unable to parse '{stringToConvert}'."
    | :? OverflowException ->
        printfn $"'{stringToConvert}' is greater than {Byte.MaxValue} or less than {Byte.MinValue}." 

    let stringToConvert = " +214 "
    try
        let byteValue = Byte.Parse(stringToConvert, CultureInfo.InvariantCulture)
        printfn $"Converted '{stringToConvert}' to {byteValue}." 
    with
    | :? FormatException ->
        printfn $"Unable to parse '{stringToConvert}'."
    | :? OverflowException ->
        printfn $"'{stringToConvert}' is greater than {Byte.MaxValue} or less than {Byte.MinValue}." 

    // The example displays the following output to the console:
    //       Converted ' 214 ' to 214.
    //       Unable to parse ' + 214 '.
    //       Converted ' +214 ' to 214.
    // </Snippet2>

let callParse3 () =
    // <Snippet3>
    // Parse value with no styles allowed.
    let style = NumberStyles.None
    let value = " 241 "
    try
        let number = Byte.Parse(value, style);
        printfn $"Converted '{value}' to {number}."
    with :? FormatException ->
        printfn $"Unable to parse '{value}'."

    // Parse value with trailing sign.
    let style = NumberStyles.Integer ||| NumberStyles.AllowTrailingSign
    let value = " 163+"
    let number = Byte.Parse(value, style)
    printfn $"Converted '{value}' to {number}."

    // Parse value with leading sign.
    let value = "   +253  "
    let number = Byte.Parse(value, style)
    printfn $"Converted '{value}' to {number}."

    // This example displays the following output to the console:
    //       Unable to parse ' 241 '.
    //       Converted ' 163+' to 163.
    //       Converted '   +253  ' to 253.
    // </Snippet3>

let callParse4 () =
    // <Snippet4>
    // Parse number with decimals.
    // NumberStyles.Float includes NumberStyles.AllowDecimalPoint.
    let style = NumberStyles.Float
    let culture = CultureInfo.CreateSpecificCulture "fr-FR"
    let value = "12,000"

    let number = Byte.Parse(value, style, culture)
    printfn $"Converted '{value}' to {number}."

    let culture = CultureInfo.CreateSpecificCulture "en-GB"
    try
        let number = Byte.Parse(value, style, culture)
        printfn $"Converted '{value}' to {number}."
    with :? FormatException ->
        printfn $"Unable to parse '{value}'."

    let value = "12.000"
    let number = Byte.Parse(value, style, culture)
    printfn $"Converted '{value}' to {number}."
    
    // The example displays the following output to the console:
    //       Converted '12,000' to 12.
    //       Unable to parse '12,000'.
    //       Converted '12.000' to 12.
    // </Snippet4>

callParse1 ()
printfn ""
callParse2 ()
printfn ""
callParse3 ()
printfn ""
callParse4 ()

File: snippets/csharp/System/Byte/Parse/parse.cs
using System;
using System.Globalization;

public class Class1
{
    public static void Main()
    {
        CallParse1();
        Console.WriteLine();
        CallParse2();
        Console.WriteLine();
        CallParse3();
        Console.WriteLine();
        CallParse4();
    }

    private static void CallParse1()
    {
        // <Snippet1>
        string stringToConvert = " 162";
        byte byteValue;
        try
        {
            byteValue = byte.Parse(stringToConvert);
            Console.WriteLine($"Converted '{stringToConvert}' to {byteValue}.");
        }
        catch (FormatException)
        {
            Console.WriteLine($"Unable to parse '{stringToConvert}'.");
        }
        catch (OverflowException)
        {
            Console.WriteLine($"'{stringToConvert}' is greater than {byte.MaxValue} or less than {byte.MinValue}.");
        }
        // The example displays the following output to the console:
        //       Converted ' 162' to 162.
        // </Snippet1>
    }

    private static void CallParse2()
    {
        // <Snippet2>
        string stringToConvert;
        byte byteValue;

        stringToConvert = " 214 ";
        try
        {
            byteValue = byte.Parse(stringToConvert, CultureInfo.InvariantCulture);
            Console.WriteLine($"Converted '{stringToConvert}' to {byteValue}.");
        }
        catch (FormatException)
        {
            Console.WriteLine($"Unable to parse '{stringToConvert}'.");
        }
        catch (OverflowException)
        {
            Console.WriteLine($"'{stringToConvert}' is greater than {byte.MaxValue} or less than {byte.MinValue}.");
        }

        stringToConvert = " + 214 ";
        try
        {
            byteValue = byte.Parse(stringToConvert, CultureInfo.InvariantCulture);
            Console.WriteLine($"Converted '{stringToConvert}' to {byteValue}.");
        }
        catch (FormatException)
        {
            Console.WriteLine($"Unable to parse '{stringToConvert}'.");
        }
        catch (OverflowException)
        {
            Console.WriteLine($"'{stringToConvert}' is greater than {byte.MaxValue} or less than {byte.MinValue}.");
        }

        stringToConvert = " +214 ";
        try
        {
            byteValue = byte.Parse(stringToConvert, CultureInfo.InvariantCulture);
            Console.WriteLine($"Converted '{stringToConvert}' to {byteValue}.");
        }
        catch (FormatException)
        {
            Console.WriteLine($"Unable to parse '{stringToConvert}'.");
        }
        catch (OverflowException)
        {
            Console.WriteLine($"'{stringToConvert}' is greater than {byte.MaxValue} or less than {byte.MinValue}.");
        }
        // The example displays the following output to the console:
        //       Converted ' 214 ' to 214.
        //       Unable to parse ' + 214 '.
        //       Converted ' +214 ' to 214.
        // </Snippet2>
    }
    private static void CallParse3()
    {
        // <Snippet3>
        string value;
        NumberStyles style;
        byte number;

        // Parse value with no styles allowed.
        style = NumberStyles.None;
        value = " 241 ";
        try
        {
            number = byte.Parse(value, style);
            Console.WriteLine($"Converted '{value}' to {number}.");
        }
        catch (FormatException)
        {
            Console.WriteLine($"Unable to parse '{value}'.");
        }

        // Parse value with trailing sign.
        style = NumberStyles.Integer | NumberStyles.AllowTrailingSign;
        value = " 163+";
        number = byte.Parse(value, style);
        Console.WriteLine($"Converted '{value}' to {number}.");

        // Parse value with leading sign.
        value = "   +253  ";
        number = byte.Parse(value, style);
        Console.WriteLine($"Converted '{value}' to {number}.");
        // This example displays the following output to the console:
        //       Unable to parse ' 241 '.
        //       Converted ' 163+' to 163.
        //       Converted '   +253  ' to 253.
        // </Snippet3>
    }

    private static void CallParse4()
    {
        // <Snippet4>
        NumberStyles style;
        CultureInfo culture;
        string value;
        byte number;

        // Parse number with decimals.
        // NumberStyles.Float includes NumberStyles.AllowDecimalPoint.
        style = NumberStyles.Float;
        culture = CultureInfo.CreateSpecificCulture("fr-FR");
        value = "12,000";

        number = byte.Parse(value, style, culture);
        Console.WriteLine($"Converted '{value}' to {number}.");

        culture = CultureInfo.CreateSpecificCulture("en-GB");
        try
        {
            number = byte.Parse(value, style, culture);
            Console.WriteLine($"Converted '{value}' to {number}.");
        }
        catch (FormatException)
        {
            Console.WriteLine($"Unable to parse '{value}'.");
        }

        value = "12.000";
        number = byte.Parse(value, style, culture);
        Console.WriteLine($"Converted '{value}' to {number}.");
        // The example displays the following output to the console:
        //       Converted '12,000' to 12.
        //       Unable to parse '12,000'.
        //       Converted '12.000' to 12.
        // </Snippet4>
    }
}
